### PR TITLE
Adding paragraph completion

### PR DESCRIPTION
**Pull request by abdrabrcom**
adding Paragraph completion suggestion as per #28

### CODE REVIEW

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletionTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  @Test
+  public void testParagraphCompletion() {
+    ParagraphCompletion completion = new ParagraphCompletion();
+    MyDocumentModel document = createModel();
+
+    List<CompletionItem> completionItems =
+        completion.collectCompletions(document, createCompletionParams());
+
+    assertEquals(2, completionItems.size());
+    // assertTrue("000-Main-Logic", completionItems.get(1).getLabel());
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletionTest.java]: Please, remove the commented code

Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletionTest.java]: Done

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/KeywordCompletion.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ String tryResolve(String label) {
 
   @Override
   protected String getSortOrderPrefix() {
-    return "2"; // Keywords should go after the variables in the completions list
+    return "3"; // Keywords should go after the variables in the completions list
```
Comment (zacanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/KeywordCompletion.java]: I think here we can add a kind of javadoc to explain each number what means

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletion.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return Optional.ofNullable(document)
+        .map(MyDocumentModel::getAnalysisResult)
+        .map(AnalysisResult::getParagraphs)
+        .orElse(Collections.emptySet());
+  }
+
+  @Override
+  String tryResolve(String label) {
+    // Not yet supported
+    return null;
```
Comment (zacanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletion.java]: throw an UnsupportedOperationException

Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletion.java]: Done

Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/completions/ParagraphCompletion.java]: this is part of the method logic and is needed, comments have been updated to provide more information

**Pull request merged**